### PR TITLE
Improve advanced search "clear" behavior for radio buttons/checkboxes

### PR DESCRIPTION
**Pull request by xrosecky**
We noticed that advanced search in EDS contains checkboxes that are not cleared when clear button is used.

### CODE REVIEW

**Comment by demiankatz** [comment]
Thanks, @xrosecky! When time permits (either later this week or early next week) I'm going to see if I can write a Mink integration test to cover this case so that we can catch future regressions. I also wonder if the very specific code to click on a particular "illustrated" option can be made more generic (it would be better to reset all radio buttons rather than just this specific one, to support future growth/customization).

**Comment by demiankatz** [comment]
...at a glance, it actually looks like repeating your newly-added line with a type of "radio" in place of the illustrated click logic might do exactly what we need. I'll try it when I have time, unless you think it's worth doing and get to it first.

File: themes/bootstrap3/js/advanced_search.js [hunk 1]
```diff
@@ -137,6 +137,7 @@ deleteGroup = function _deleteGroup(group) {
 $(document).ready(function advSearchReady() {
   $('.clear-btn').click(function clearBtnClick() {
     $('input[type="text"]').val('');
+    $('input[type="checkbox"]').prop("checked", false);
```
Comment (EreMaijala) [inline themes/bootstrap3/js/advanced_search.js]: EDS can have checkboxes that are checked by default. I don't think it's appropriate to clear them. All checkboxes should rather be reset to the initial state.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: themes/bootstrap3/js/advanced_search.js (hunk 1 above)
Comment (xrosecky) [reply themes/bootstrap3/js/advanced_search.js]: So should I store the initial state as data attribute and use it for reseting the checkbox to the initial state?

Comment (EreMaijala) [reply themes/bootstrap3/js/advanced_search.js]: Yes, I believe that's the clearest way to do it.

Comment (demiankatz) [reply themes/bootstrap3/js/advanced_search.js]: Thanks for adding this, @xrosecky -- so I think now the next step is adding the "reset-checked" attribute where it is needed. I'll look into this next week when I have a bit more time, unless somebody else gets there first. :-)

Comment (demiankatz) [reply themes/bootstrap3/js/advanced_search.js]: Okay, please take a look at ed5453e8b093d5a1a18348ee5bcacf577201acf6 -- this uses the new reset-checked attribute for both EDS checkboxes and Solr radio buttons. I think this is probably safe to merge now; does anyone have any objections or further suggestions?

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Looks good to me otherwise, but now that I look at it, it occurs to me that something like 'data-initially-checked' might be more intuitive, as 'reset-checked' might be misunderstood for "checked needs to be reset [to false]".

**Comment by demiankatz** [comment]
Thanks, @EreMaijala. `data-checked-by-default` might also be a potential name for this. @xrosecky, do you have any preferences? I'm happy to make the change once we agree on the best name.

File: themes/bootstrap3/js/advanced_search.js (hunk 1 above)
Comment (xrosecky) [reply themes/bootstrap3/js/advanced_search.js]: It is OK for me.

Comment (demiankatz) [reply themes/bootstrap3/js/advanced_search.js]: Thanks, @xrosecky. Do you have any opinions about the discussion below about renaming `reset-checked` to `initially-checked` or `checked-by-default` for clarity?

Comment (xrosecky) [reply themes/bootstrap3/js/advanced_search.js]: Data attribute `checked-by-default ` seems better to me.

Comment (demiankatz) [reply themes/bootstrap3/js/advanced_search.js]: Thanks, @xrosecky, I have made the change and will merge this as soon as CI passes.

**Review by demiankatz — APPROVED** [review]

**Pull request merged**